reVX/hybrid_stats/hybrid_stats.py

```diff
@@ -0,0 +1,1588 @@
+            Number of sites to extract on each worker, by default 1000
+        lat_lon_only : bool, optional
+            Only append lat, lon coordinates to stats, by default True
+        out_path : str, optional
+            Directory, .csv, or .json path to save statistics too,
+            by default None
+
+        Returns
+        -------
+        out_stats : pandas.DataFrame
+            DataFrame of resource statistics
+        """
+        hybrid_stats = cls(solar_h5, wind_h5, statistics=statistics,
+                           res_cls=res_cls)
+        out_stats = hybrid_stats.compute_stats(
+            'cf_profile', diurnal=diurnal, doy=doy, month=month,
+            combinations=combinations, max_workers=max_workers,
+            sites_per_worker=sites_per_worker,
+            lat_lon_only=lat_lon_only)
+        if out_path is not None:
+            hybrid_stats.save_stats(out_stats, out_path)
+
+        return out_stats
+
+
+class HybridCrossCorrelation(HybridStats):
+    """
+    Compute the temporal cross correlations for co-located wind and solar
+    generation
+    """
+
+    def __init__(self, solar_h5, wind_h5, res_cls=Resource):
+        """
+        Parameters
+        ----------
+        solar_h5 : str
+            Path to solar h5 file(s)
+        wind_h5 : str
+            Path to wind h5 file(s)
+        res_cls : Class, optional
+            Resource class to use to access res_h5, by default Resource
+        """
+        self._solar_h5 = solar_h5
+        self._wind_h5 = wind_h5
+        self._res_cls = res_cls
+        self._stats = None
+
+        out = self._pre_flight_check()
+        self._meta, self._time_index = out[:2]
+        self._solar_time_slice, self._wind_time_slice = out[2:]
+
+    def __repr__(self):
+        msg = ('Computing cross-correlations between {} and {}'
+               .format(self.solar_h5, self.wind_h5))
+
+        return msg
+
+    @staticmethod
+    def cross_correlation(solar_data, wind_data, m):
+        """
+        Compute the cross-correlation between solar and wind time-series data
+        with time-lag m
+
+        Parameters
+        ----------
+        solar_data : ndarray
+            Time-series solar data
+        wind_data : ndarray
+            Time-series wind data
+        m : int
+            Integer shift between solar and wind time-series, is pass directly
+            to np.roll, so it will be the number of time-steps that are
+            shifted. The lag time will be m * dt where dt is the time-step
+            size.
+
+        Returns
+        -------
+        corr : ndarray
+            Cross-correlation coefficient for each solar, wind site pair
+        """
+        solar_u = solar_data.mean(axis=0)
+        solar_s = solar_data.std(axis=0)
+        solar_data = np.abs((solar_data - solar_u) / solar_s)
+
+        wind_u = wind_data.mean(axis=0)
+        wind_s = wind_data.mean(axis=0)
+        wind_data = np.roll(wind_data, m, axis=0)
```

Comment:
Why do you roll one of the datasets? Is this to correlate daytime solar with 12-hour offset nighttime wind (for example)?